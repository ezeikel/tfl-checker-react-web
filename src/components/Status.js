import React, { useState, useEffect } from "react";
import styled from "styled-components";
import Card from "./Card";

const Lines = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  grid-gap: 32px;
  justify-content: space-between;
  background-color: var(--color-white);
  padding: 32px;
  border-radius: 4px;
`;

const Status = () => {
  const [status, setStatus] = useState([]);

  useEffect(() => {
    async function fetchData() {
      const data = await fetch(
        "https://api.tfl.gov.uk/line/mode/tube,overground,dlr,tflrail/status?app_id=1b83c22c&app_key=e5c7b582d0f72a04add248393e939cf5",
      );
      const json = await data.json();

      setStatus(json);
    }
    fetchData();
  }, []);

  const renderLines = () =>
    status.map(line => <Card key={line.id} line={line} />);

  return <Lines>{renderLines()}</Lines>;
};

export default Status;
