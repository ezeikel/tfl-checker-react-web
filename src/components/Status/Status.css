.tfl-lines {
  display: flex;
  flex-direction: column;
}

.line > span {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0.5rem 1rem;
}

.line > span:first-of-type {
  color: #fff;
}

.line > span:nth-of-type(2) {
  background-color: #fff;
}

.bakerloo {
  background-color: var(--bakerloo-color);
}

.central {
  background-color: var(--central-color);
}

.circle {
  background-color: var(--circle-color);
}

.district {
  background-color: var(--district-color);
}

.hammersmith-city {
  background-color: var(--hammersmith-city-color);
}

.jubilee {
  background-color: var(--jubilee-color);
}

.metropolitan {
  background-color: var(--metropolitan-color);
}

.northern {
  background-color: var(--northern-color);
}

.piccadilly {
  background-color: var(--picadilly-color);
}

.victoria {
  background-color: var(--victoria-color);
}

.waterloo-city {
  background-color: var(--waterloo-city-color);
}