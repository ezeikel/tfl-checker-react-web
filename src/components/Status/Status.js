import React, { Component } from 'react';

class Status extends Component {
    render() {
        return (
            <div>Status component works!</div>
        )
    }
};

export default Status;