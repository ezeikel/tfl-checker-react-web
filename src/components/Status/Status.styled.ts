/* eslint-disable import/prefer-default-export */

import styled from "styled-components";

export const Lines = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  grid-gap: 32px;
  justify-content: space-between;
  background-color: var(--color-white);
  padding: 32px;
  border-radius: 4px;
`;
