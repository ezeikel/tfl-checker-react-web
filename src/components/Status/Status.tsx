import { useState, useEffect } from "react";
import Card from "../Card/Card";
import { Lines } from "./Status.styled";

const Status = () => {
  const [status, setStatus] = useState([]);

  useEffect(() => {
    async function fetchData() {
      const data = await fetch(
        "https://api.tfl.gov.uk/line/mode/tube,overground,dlr,tflrail/status?app_id=1b83c22c&app_key=e5c7b582d0f72a04add248393e939cf5",
      );
      const json = await data.json();

      setStatus(json);
    }
    fetchData();
  }, []);

  const renderLines = () =>
    status.map((line) => <Card key={line.id} line={line} />);

  return <Lines>{renderLines()}</Lines>;
};

export default Status;
