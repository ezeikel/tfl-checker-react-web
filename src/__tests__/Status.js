import React from 'react';
import { render } from '@testing-library/react';
import Status from '../components/Status';

test('renders without crashing', () => {
  const { container } = render(
    <Status />
  );
  expect(container).toMatchSnapshot();
});