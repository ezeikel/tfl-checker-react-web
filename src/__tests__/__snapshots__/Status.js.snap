// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="sc-AxhCb kicRlZ"
  />
</div>
`;
